internal/validation/openapi.go

```diff
@@ -296,5 +299,37 @@ func Validate(ctx context.Context, schema []byte, schemaPath string, limits *Out
 		}
 	}
 
+	searchDirs := map[string]bool{
+		workingDir: true,
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err == nil {
+		searchDirs[homeDir] = false
+	}
+
+	outputDir := filepath.Join(os.TempDir(), "speakeasy")
+
+	for dir, allowRecursive := range searchDirs {
+		res, err := workspace.FindWorkspace(dir, workspace.FindWorkspaceOptions{
+			Recursive: allowRecursive,
+		})
+		if err != nil {
+			continue
+		}
+		outputDir = filepath.Join(res.Path, "temp")
+	}
+
+	_ = os.MkdirAll(outputDir, 0o755)
```

Comment:
do we not care about this error?

Reply:
I have seen this fail rarely (and when it does its because it already exists) and if this fails for other reasons then the following writing of a file will also fail for those reasons